weatherAplication/Modules/Home/View/HomeView.swift

```diff
@@ -0,0 +1,101 @@
+import SnapKit
+import UIKit
+
+class HomeView: UIView {
+    private let kLogoWidth: CGFloat = 250
+    private let kLogoHeight: CGFloat = 250
+    private let kSideMargin: CGFloat = 10
+    private let kTopMargin: CGFloat = 20
+    private let kBottomMargin: CGFloat = 15
+    private let kTableBottomMargin: CGFloat = 50
+    private let aplicationImage = R.image.weather_symbol()
+    private let logoImageView = UIImageView()
+    private let aplicationNameLabel = UILabel()
+    internal let searchButton = SearchButton()
+    private let aplicationNameString: String = "MyWeather"
+    internal let lastCityNamesTableView = UITableView()
+
+    private var viewModel: HomeViewModelContract
```

Comment:
Do view nie przekazujemy viewModelu, bo zupełnie nie ma takiej potrzeby. Jeśli chcemy zasetupować działanie np. buttona to robimy to z poziomu viewControllera. I też wszystkie obiekty np. przyciski, tabele, labelki itp nie powinny być private ani internal. View tak naprawdę ma być tylko klasą która zawiera to co się wyświetla. Wszelkie działania na widoku są kontrolowane przez viewController, a logika jest zawarta w viewModelu

Reply:
dodam, ze zgodnie z MVVM viewModel moze byc w widoku, natomiast u nas rozdzielamy dodatkowo odpowiedzialnosc widoku miedzy View i ViewController, wiec wychodzi w sumie MVVCVM. tak jak Andrzej napisal, View ma byc mozliwie glupi, wszystkie dane do niego przekazywane i dzialania sa wykonywane przez VC